lib/checkclass.cpp

```diff
@@ -1551,7 +1565,42 @@ bool CheckClass::hasAllocation(const Function *func, const Scope* scope) const
     return false;
 }
 
-bool CheckClass::hasAssignSelf(const Function *func, const Token *rhs)
+bool CheckClass::isInverseAssignmentTest(const Token *tok)
+{
+	bool res = true;
+	for (const Token *itr = tok; itr; itr=itr->astParent()) {
+		if (itr->str() == "!=" && (itr->astOperand1()->str() == "true" || itr->astOperand2()->str() == "true")) {
+			res = !res;
+		}
+		else if (itr->str() == "!=" && (itr->astOperand1()->str() != "false" && itr->astOperand2()->str() != "false")) {
```

Comment:
In general I prefer that we use valueflow instead of hardcoded constants. A `false` token has the known integer value `0`..

     (... astOperand1()->hasKnownIntValue() && .. astOperand1()->getKnownIntValue() != 0 )

.. and a `true` token has the value `1` ..

Reply:
Done